app/controllers/submit.js

```diff
@@ -184,6 +184,15 @@ export default Ember.Controller.extend(BasicsValidations, NodeActionsMixin, Tagg
         return this.get('isAdmin') && !(this.get('node.registration'));
     }),
 
+    renderMathJax: Ember.observer('uploadValid', 'node.title', function() {
+        Ember.run.next(this, function() {
+            //node.title gets changed on every character input, so debounce this.
+            Ember.run.debounce(this, function() {
```

Comment:
It seems that you really only want to render mathjax once- when the form section switches from input mode ("here are some textboxes") to preview mode ("here is what it would look like").

In that case, you may want a one-time typeset operation to be called as part of an action- observers might not be the mechanism you want at all. It shouldn't be necessary to debounce an operation on a field that only displays when the user is not able to edit the underlying value.